src/plot_api/plot_api.js

```diff
@@ -2313,6 +2322,17 @@ exports.react = function(gd, data, layout, config) {
         var restyleFlags = diffData(gd, oldFullData, newFullData, immutable);
         var relayoutFlags = diffLayout(gd, oldFullLayout, newFullLayout, immutable);
 
+        // TODO: how to translate this part of relayout to Plotly.react?
+        // // Setting width or height to null must reset the graph's width / height
+        // // back to its initial value as computed during the first pass in Plots.plotAutoSize.
+        // //
+        // // To do so, we must manually set them back here using the _initialAutoSize cache.
+        // if(['width', 'height'].indexOf(ai) !== -1 && vi === null) {
+        //     fullLayout[ai] = gd._initialAutoSize[ai];
+        // }
```

Comment:
On closer inspection I *think* I understand what this is about, but I don't think it's actually the correct behavior, for example if `gd` was resized external to plotly.js it wouldn't make sense to revert to when the plot was first drawn. I'm not quite sure how to handle this though.

Reply:
:grimacing: auto-sizing always has a way to make my brain hurt.

Referencing https://github.com/plotly/plotly.js/pull/635 which did a lot of damage the streambed :face_with_head_bandage:

Reply:
Oy, I think we've all fought with that one at various points... If you don't mind, I'm going to leave this as it is now - with the TODO here to remind us about it if a related bug pops up, but without trying to implement anything around it. We have some great tests in [config_test](https://github.com/plotly/plotly.js/pull/2437/commits/9f774c0b650b8f913b00337afc1bdcf3735e7c04#diff-42c68411da55979155205ae2842e992fR87) that I extended to make sure that all of *those* test cases work the same in `relayout` and `react`, I'd like to just freeze it until we have more test cases to look at.